src/main/java/com/mineboundteam/minebound/magic/OffensiveSpells/NecroticOffensiveSpell.java

```diff
@@ -0,0 +1,170 @@
+package com.mineboundteam.minebound.magic.OffensiveSpells;
+
+import java.util.List;
+
+import com.mineboundteam.minebound.MineBound;
+import com.mineboundteam.minebound.capabilities.ArmorNBTHelper;
+import com.mineboundteam.minebound.capabilities.PlayerSelectedSpellsProvider;
+import com.mineboundteam.minebound.capabilities.PlayerSelectedSpellsProvider.PrimarySpellProvider.PrimarySelected;
+import com.mineboundteam.minebound.capabilities.PlayerSelectedSpellsProvider.SecondarySpellProvider.SecondarySelected;
+import com.mineboundteam.minebound.config.IConfig;
+import com.mineboundteam.minebound.item.armor.ArmorTier;
+import com.mineboundteam.minebound.magic.ActiveSpellItem;
+import com.mineboundteam.minebound.magic.MagicType;
+import com.mineboundteam.minebound.magic.SpellType;
+
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.food.FoodData;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.TooltipFlag;
+import net.minecraft.world.level.Level;
+import net.minecraftforge.common.ForgeConfigSpec;
+import net.minecraftforge.common.ForgeConfigSpec.DoubleValue;
+import net.minecraftforge.common.ForgeConfigSpec.IntValue;
+import net.minecraftforge.event.entity.living.LivingDamageEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.event.config.ModConfigEvent;
+
+
+@Mod.EventBusSubscriber(modid = MineBound.MOD_ID)
+public class NecroticOffensiveSpell extends ActiveSpellItem {
+    public static DoubleValue FOOD_REDUCTION;
+
+    private final int manaCost;
+    private final double damageBoost;
+    private final double vampirePercent;
+
+    public final static String ACTIVE_TAG = MineBound.MOD_ID + ".necrotic_offensive.active";
+    public final static String TICK_TAG = MineBound.MOD_ID + ".necrotic_offensive.ticks";
+
+    public NecroticOffensiveSpell(Properties properties, NecroticOffensiveSpellConfig config) {
+        super(properties, config.LEVEL, MagicType.NECROTIC, SpellType.OFFENSIVE);
+        manaCost = config.MANA_COST.get();
+        damageBoost = config.DMG_BOOST.get();
+        vampirePercent = config.HEALTH_AMT.get();
+    }
+
+    @Override
+    public void use(ItemStack stack, Level level, Player player) {
+        stack.getOrCreateTag().putBoolean(ACTIVE_TAG, true);
+    }
+
+    @Override
+    public void onUsingTick(ItemStack stack, Level level, Player player) {
```

Comment:
You might be able to update this function in `ActiveSpellItem` to take the tick count. I realized I probably should have done that too late so I was going to make that change once it was necessary, but you might benefit from it here and not need to use NBT to count the ticks by just checking `count % 20`.

Reply:
I think it's the tick count. At least, that's how our networking was using that argument. But you can test to see if it actually is.

Reply:
It's coming in as zero from the network

Reply:
Alright, I'll probably take a pass at it when I eventually decide to finish Fire Offensive since it will be useful there.